### ISSUE TITLE
ActiveRecord left joins out of order

### ISSUE THREAD

**Issue by ryan-plated** [rails/rails#32598]
### Steps to reproduce

Combine sql left join with a arel join causes wrong order

https://gist.github.com/ryan-plated/cae50e8eceb621f8a76d41f2e6d1b5ec

### Expected behavior
should keep order

### Actual behavior
goes out of order

### System configuration
**Rails version**: master

**Ruby version**: 2.5.0

**Comment by ryan-plated** [comment]
I am not sure what you are trying to communicate. I read the post and it says outer joins order does matter.

**Comment by rafaelfranca** [comment]
Could you explain better the expected behavior and the actual behavior? "should keep order" doesn't tell anything to me. Order of what?

**Comment by rafaelfranca** [comment]
I think it is related to https://github.com/rails/rails/pull/15461

**Comment by ryanong** [comment]
YEA that is exactly it! I'll gladly pick up the stale PR! I just didn't even know where to start.

**Comment by heaven** [comment]
Here is our example that raises the `Unknown column` error:

```ruby
scope :expired, ->(service, time = 1.month) {
  distinct.
    left_joins(:timestamps).
    joins("LEFT JOIN timestamp_kinds...")
}
```

**Comment by dgmora** [comment]
Is there an active PR/issue for this (still existing) issue? #15461 is closed and that one links to #24281, which is kinda related but is not the same, I think

**Comment by hassanrehman** [comment]
+1 for PR

**Comment by mateuscruz** [comment]
I'm willing to create a PR for this. I ran into this issue earlier today.

**Comment by khiav223577** [comment]
Did this bug be fixed? I ran into this issue, too.

**Comment by kamipo** [comment]
Fixed by #35864 in Rails 6.0.
